data_steward/cdr_cleaner/cleaning_rules/identifying_field_suppression.py

```diff
@@ -0,0 +1,181 @@
+
+# identifying fields to be suppressed
+fields = [
+    'month_of_birth', 'day_of_birth', 'location_id', 'provider_id',
+    'care_site_id'
+]
+
+# negative age at recorded time in table
+ID_FIELD_SUPPRESSION_QUERY = JINJA_ENV.from_string("""
+SELECT
+  *
+  REPLACE({{replace_statement}})
+FROM
+  `{{project_id}}.{{dataset_id}}.{{table}}`
+""")
+
+# replace string to use in query
+REPLACE_STRING = JINJA_ENV.from_string("""
+{{suppression_statement}} AS {{field}}
+""")
+
+
+class IDFieldSuppression(BaseCleaningRule):
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Returns queries to null the data in identifying fields for all OMOP common data model tables.'
+        )
+
+        super().__init__(
+            issue_numbers=JIRA_ISSUE_NUMBERS,
+            description=desc,
+            affected_datasets=[cdr_consts.COMBINED],
+            affected_tables=CDM_TABLES,
+            project_id=project_id,
+            dataset_id=dataset_id,
+            sandbox_dataset_id=sandbox_dataset_id,
+            depends_on=[TableSuppression]
+        )  # table_suppression.py module will handle identifying fields in provider, care_site, location
+
+    def get_query_specs(self, *args, **keyword_args) -> query_spec_list:
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+        queries = []
+        current_dir = os.getcwd()
+        # loop through tables to create replace statements
+        for table in CDM_TABLES:
+            # change to root directory to access fields files
+            root_dir = os.path.dirname(
+                os.path.dirname(os.path.abspath(__file__)))
+            os.chdir(root_dir)
+
+            # create replace statement by looping through schema
+            with open(f'../resource_files/fields/{table}.json') as f:
+                schema = json.load(f)
+                statements = []
+                for item in schema:
+                    if item['name'] in fields:
+                        if item['mode'] == 'nullable':
+                            value = 'NULL'
+                        elif item['type'] == 'integer':
+                            value = 0
+                        elif item['type'] == 'string':
+                            value = ''
+                        suppression_statement = REPLACE_STRING.render(
+                            suppression_statement=value, field=item['name'])
```

Comment:
Changing the directory feels a bit off. You can make use of the helper functions in `utils.bq` for this.
```suggestion
          for table in self.affected_tables:
             schema = bq.get_table_schema(table)
             statements = []
             for item in schema:
                 if item.name in fields:
                     if item.mode == 'nullable':
                         value = 'NULL'
                     elif item.field_type == 'integer':
                         value = 0
                     elif item.field_type == 'string':
                         value = ''
                     else:
                         raise RuntimeError(
                             f"Required field {item.name} needs to be integer or string type to be replaced"
                         )
                     suppression_statement = REPLACE_STRING.render(
                         suppression_statement=value, field=item.name)
```

Reply:
this is much better, thank you for that suggestion!